mlir/lib/Target/CppOutput/gridwise_convolution_implicit_gemm_v4r4_gen_xdlops.cpp

```diff
@@ -28,7 +28,6 @@ using namespace mlir;
 
 namespace {
 // result string to keep C++ source / header / flags emission.
-std::string resultStr;
```

Comment:
I don't mind about you making changes to `CppOutput` folder is general because cpp backend is going to be deprecated.

However, Making `std::string` to be local will likely make the existing cpp output unit test seg fault. As far as I understand, `stringRef` does not own the memory, whereas it is owned by the `std::string`, created on stack. Once the related API function are done, the `resultStr` will be destroyed and the client will have a dangling `stringRef` pointing trash memory.

Reply:
Yes, that is a problem. I moved it because mlir/lib/CMakeLists.txt:2 doesn't like globals in some compilers:

>  add_flag_if_supported("-Werror=global-constructors" WERROR_GLOBAL_CONSTRUCTOR)

I will just return the string since it is going away. Sound good?

Reply:
Sounds good to me